src/pyramid/i18n.py

```diff
@@ -353,18 +348,28 @@ def dngettext(self, domain, singular, plural, num):
         """
         return self._domains.get(domain, self).ngettext(singular, plural, num)
 
-    def ldngettext(self, domain, singular, plural, num):
-        """Like ``lngettext()``, but look the message up in the specified
-        domain.
-        """
-        return self._domains.get(domain, self).lngettext(singular, plural, num)
-
     def dungettext(self, domain, singular, plural, num):
         """Like ``ungettext()`` but look the message up in the specified
         domain.
         """
         return self._domains.get(domain, self).ngettext(singular, plural, num)
 
+    if sys.version_info < (3, 11):
+
+        def ldgettext(self, domain, message):
+            """Like ``lgettext()``, but look the message up in the specified
+            domain.
+            """
+            return self._domains.get(domain, self).lgettext(message)
+
+        def ldngettext(self, domain, singular, plural, num):
+            """Like ``lngettext()``, but look the message up in the specified
+            domain.
+            """
+            return self._domains.get(domain, self).lngettext(
+                singular, plural, num
+            )
+
```

Comment:
Looking at the comments on https://github.com/python/cpython/issues/77891 these functions should are only useful on Python 2. Since we dropped support for Python 2 these functions should be dropped too.

I do not think it is useful to have different behavior pre 3.11 and to keep these functions around.

As it stands I would prefer to see this just removed entirely as they do not do the right thing on Py3 anyway.

I defer to @mmerickel for the final decision on this though.

Reply:
I'll switch to removing them entirely. They shouldn't be useful to anybody, any more.